doc(policies): add policy description in tractusx-edc (#856)
eclipse-tractusx/tractusx-edc: docs/policies/policies.md

| -------- | -------- |
| Policy | A group of one or more Rules
| Rule | An abstract concept that represents the common characteristics of Permissions, Prohibitions, and Duties
| Asset | A rsource or a collection of resources that are the subject of a Rule
| Action | An operation on an Asset
| Permission | The ability to exercise an Action over an Asset
| Prohibition | The inability to exercise an Action over an Asset
| Constraint | A boolean/logical expression that refines an Action and Asset collection or the conditions applicable to a Rule.

## Policies in Catena-X

A Catena-X policy uses the ODRL Information Model as a semantic model. Theoretically, a policy can be constructed individually using this ODRL Information Model. Catena-X does not yet use all the possibilities that ODRL provides, so the next example will show how a Catena-X policy is basically structured.

### Policy Template Example with @context: `"http://www.w3.org/ns/odrl.jsonld"`

    {
      "@context": {},
      "@type": "PolicyDefinitionRequest",
      "@id": "{{POLICY_ID}}",
      "policy": {
        "@type": "Set",
        "@context": "http://www.w3.org/ns/odrl.jsonld",
        "permission": [
          {

>             "action": "use",

Reviewer: As mentioned by @matgnt, the other examples show `"@type": "LogicalConstraint"` which is missing here. The introductory sentence mentions:
> so the next example will show how a Catena-X policy is basically structured.

Therefore, this should be consistent in the whole documentation to avoid confusions.

Author: I added the type

            "constraint": [
              {
                "leftOperand": {
                  "@value": "<field>"
                },
                "operator": "eq",
                "rightOperand": "<value>"
              }
            ]
          }
        ]
      }

>     }

Reviewer 2: Please put this example into code fences

Author: done

| policy permission "constraint" | A boolean/logical expression that refines an Action and Party/Asset collection or the conditions applicable to a Rule. The leftOperand instances MUST clearly be defined to indicate the semantics of the Constraint. Catena-X will use the **left operand** of a *constraint* to associate a specific verifiable credential (VC). The left operand value will correspond to the form: [VC type].[subtype]. **Right Operand:** The rightOperand is the value of the Constraint that is to be compared to the leftOperand. All possible `Catena-X constraints` can be found here:  [https://github.com/eclipse-tractusx/ssi-docu](https://github.com/eclipse-tractusx/ssi-docu/blob/main/docs/architecture/cx-3-2/edc/policy.definitions.md)

### Policy Template Example with @context: `"odrl": "http://www.w3.org/ns/odrl/2/"`

There is another possibility of representation: in the example above, `"@context": "http://www.w3.org/ns/odrl.jsonld"` was defined, so it is not necessary to always write `odlr:xy`. If this is not defined, the policy must look like this:

>         {

Reviewer 2: Please put this json obeject into code fences and fix the indentations.

Author: done

        "@context": {
            "odrl": "http://www.w3.org/ns/odrl/2/"
        },
        "@type": "PolicyDefinitionRequest",
        "@id": "{{POLICY_ID}}",
        "policy": {
            "@type": "odrl:Set",
            "odrl:permission": [
            {
                "odrl:action": {
                "@id": "odrl:use"
                },
                "odrl:constraint": [
                {
                    "odrl:leftOperand": {
                    "@value": "<field>"
                    },
                    "odrl:operator": {
                    "@id": "odrl:eq"
                    },
                    "odrl:rightOperand": "<value>"
                }
                ]
            }
            ]
        }
        }

### Policies & Verifiable Credenitals (VC)

#### General Information

Catena-X uses policies to determine access to and use of data. The policies refer to verifiable credentials (VC) that are stored in the Managed Identity Wallet (MIW). Catena-X uses the principle of self-sovereign identity (SSI) here.

> >ℹ️ Detailed Information about SSI, VC and MIW can be found here:

Reviewer 2: Why should this be a quote?

```suggestion
ℹ️ Detailed Information about SSI, VC and MIW can be found here:

```

- [https://github.com/eclipse-tractusx/ssi-docu/tree/main/docs/architecture/cx-3-2](https://github.com/eclipse-tractusx/ssi-docu/tree/main/docs/architecture/cx-3-2) (for release 23.09)
- [https://github.com/eclipse-tractusx/ssi-docu/tree/main/docs/architecture/cx-3-3](https://github.com/eclipse-tractusx/ssi-docu/tree/main/docs/architecture/cx-3-3) (for release 23.12)

The key architectural principle underlying this specification is that policy definitions must be decoupled from their corresponding VC schema. Namely, the specific **constraints** and shape of the VC schema must not be reflected in the policy definition. This allows VC schemas to be altered without impacting policy definitions.

#### Membership Constraint Example

The main difference between the various Catena-X policies is their `constraints`. For this reason, only the constraint examples are shown below instead of always showing the entire policy.

To understand what the connection between policy and VC looks like, let's take a look at the following example ([Source](https://github.com/eclipse-tractusx/ssi-docu/blob/main/docs/architecture/cx-3-2/edc/policy.definitions.md#1-membership-constraint)):

>     {

Reviewer 2: Code fences and indentation alignment please :-)

Author: done

      "constraint": {
        "leftOperand": "Membership",
        "operator": "eq",
        "rightOperand": "active"
      }
    }

An associated VC could look like this ([Source](https://github.com/eclipse-tractusx/ssi-docu/blob/main/docs/architecture/cx-3-2/3.%20Verifiable%20Credentials/Structure%20and%20Formats.md)):

>     {

Reviewer 2: Code fences and indentation alignment please :-)

Author: done

| "@context" | An array of context URLs specifying the context for interpreting the credential |
| "type" | An array indicating the type of the credential, including "VerifiableCredential" and "MembershipCredentialCX". |
| "issuanceDate" | The date and time when the credential was issued (in UTC format) |
| "expirationDate" | The date and time when the credential expires (in UTC format) |
| "issuer" | The issuer of the credential, identified by a decentralized identifier (did) associated with the operating environment. |
| "credentialSubject" | An object containing information about the subject of the credential |
| credentialSubject "type" | The type of the credential subject, which is "MembershipCredential" |
|  credentialSubject "holderIdentifier" | A unique identifier for the holder of the credential (in this case, "bpn") |
|  credentialSubject "memberOf" | The organization or group the holder is a member of (in this case, "Catena-X") |
| credentialSubject "status" | The status of the credential, indicating it is currently "Active"
| credenitalSubject "startTime" | The date and time when the membership started (in UTC format)

### Specific Catena-X Constraints

> >ℹ️ Detailed Information about all possible `Catena-X constraints` can be found here:

Reviewer 2: Why should this be a quote?

```suggestion
ℹ️ Detailed Information about all possible `Catena-X constraints` can be found here:

```

Author: Corrected

- [https://github.com/eclipse-tractusx/ssi-docu](https://github.com/eclipse-tractusx/ssi-docu/blob/main/docs/architecture/cx-3-2/edc/policy.definitions.md)

### Access & Usage Policies

In Catena-X, a distinction is made between `Access` and `Usage` Policies.

> >ℹ️ Note: `The terms:"usage policy" and "contract policy" are used synonymously!`

Reviewer 2: ```suggestion

> ℹ️ Note: `The terms "usage policy" and "contract policy" are used synonymously!`

```

- **access policy:** determines whether a particular consumer is offered an asset or not. For example, we may want to restrict certain assets such that only consumers within a particular geography can see them. Consumers outside that geography wouldn't even have them in their catalog.
- **usage policy or contract policy:** determines the conditions for initiating a contract negotiation for a particular asset. Note that does not automatically guarantee the successful creation of a contract, it merely expresses the eligibility to start the negotiation.

> `Access and Usage Policies are not distinguished by a special semantic but by the time of the check.`

Reviewer 2: This isn't any kind of code, therefor I would remove the single code fences. If it's an important statement, you can either make it bold or put it into a quote

```suggestion
The Access and Usage Policies are not distinguished by any special semantics, but rather by the time at which they are checked.

```

Author: Corrected

To explain this in more detail, it is necessary to understand how data exchange via EDC works in Catena-X. If you want to provide data, you must create a contract definition. This looks like this, for example:

>         curl -X POST "${BOB_DATAMGMT_URL}/data/contractdefinitions" \

Reviewer 2: Code fences and indentation alignment please :-)

            --header 'X-Api-Key: password' \
            --header 'Content-Type: application/json' \
            --data '{
                        "id": "1",
                        "criteria": [
                            {
                                "operandLeft": "asset:prop:id",
                                "operator": "=",
                                "operandRight": "1"
                            }
                        ],
                        "accessPolicyId": "1",
                        "contractPolicyId": "1"
                    }' \
            -s -o /dev/null -w 'Response Code: %{http_code}\n'

As you can see, the contract definition determines which access or usage policy is referred to. This is done via the respective id's.

> >ℹ️ More detailed information about the data transfer in Catena-X can be found [here](https://github.com/eclipse-tractusx/tractusx-edc/blob/main/docs/samples/Transfer%20Data.md).

Reviewer 2: ```suggestion
> ℹ️ More detailed information about the data transfer in Catena-X can be found [here](https://github.com/eclipse-tractusx/tractusx-edc/blob/main/docs/samples/Transfer%20Data.md).

```

### EDC Policy Playground

> >
> >ℹ️ Policies can be created individually with the following tool:

Reviewer 2: ```suggestion

Policies can be created individually with the following tool:
```

- Online Version: [Link](https://eclipse-tractusx.github.io/tutorial-resources/policy-playground/)
- Local Version: [Link](https://github.com/eclipse-tractusx/tutorial-resources/tree/main/edc-policy-playground)

> >

Reviewer 2: ```suggestion

```

Author: Corrected

### Exemplary scenarios

> For the following Scenarios, we assume there is a **Partner 1 (provider)** who wants to provide Data for **Partner 2 (consumer)**.

Reviewer 2: ```suggestion
For the following Scenarios, we assume there is a **Partner 1 (provider)** who wants to provide Data for **Partner 2 (consumer)**

```

- Partner 1 (provider) has the Business-Partner-Number BPN12345.
- Partner 2 (consumer) has the Business-Partner-Number BPN6789.

Partner 2 (consumer) signed the **Traceability Framework Agreement** and followed all the necessary steps that the Credential appears within Partner 2s identity.

So for this example, if Partner 2 does a catalogue request to Partner 1, the following summary credential is provided to Partner 1:

    {
        "@context": [
            "https://w3id.org/2023/catenax/credentials/summary/v1"
        ],
        "id": "<credential uid>",
        "type": [
            "VerifiableCredential",
            "SummaryCredential"
        ],
        "issuer": "<did:web:issuer>",
        "issuanceDate": "2023-06-02T12:00:00Z",
        "expirationDate": "2022-06-16T18:56:59Z",
        "credentialSubject": {
            "id": "<did:web:subject>",
            "holderIdentifier": "BPN6789",
            "items": [
            "MembershipCredential",
            "TraceabilityCredential"
            ],
            "contractTemplates": "https://public.catena-x.org/contracts/"
        }

>     }

Reviewer 2: Code fences and indentation alignment please :-)

Author: done

#### Scenario 1

Partner 1 wants to create an Access Policy, that Partner 2 can only receive the Contract offer if its BPN matches AND if it has the Traceability Credential (Traceability Contract Signed). If one of those is missing, Partner 2 won't receive a Contract Offer.

##### Partner 1 - Access Policy Example

>     {

Reviewer 2: Code fences and indentation alignment please :-)

Author: done

        "policy": {
            "@type": "Policy",
            "odrl:permission": [
                {
                    "odrl:action": "USE",
                    "odrl:constraint": {
                        "@type": "LogicalConstraint",
                        "odrl:and": [
                            {
                                "@type": "Constraint",
                                "odrl:leftOperand": "BusinessPartnerNumber",
                                "odrl:operator": {
                                    "@id": "odrl:eq"
                                },
                                "odrl:rightOperand": "{{BPN6789}}"
                            },
                            {
                                "@type": "Constraint",
                                "odrl:leftOperand": "FrameworkAgreement.traceability",
                                "odrl:operator": {
                                    "@id": "odrl:eq"
                                },
                                "odrl:rightOperand": "active"
                            }
                        ]
                    }
                }
            ]
        }
    }

> Be aware, the credential within the Summary Credential is called "TraceabilityCredential", but as the EDC does a mapping, the EDC Access Policy Entry needs to be named "FrameworkAgreement.traceability"!

Reviewer 2: ```suggestion
Be aware, the credential within the Summary Credential is called `TraceabilityCredential`, but as the EDC does a mapping, the EDC Access Policy Entry needs to be named `FrameworkAgreement.traceability`!

```

Author: Corrected

##### Desired Outcome

Partner 2 receives the Contract Offer because the BPNs are matching and he owns the Traceability Credential.

#### Scenario 2

Partner 1 wants to create an Access Policy, that Partner 2 can receive the Contract Offer if its BPN matches. But a Contract Agreement should only be created if Partner 2 also signed the Traceability Framework Agreement. So in this case, Partner 2 should receive the Contract Offer in the first place, regardless if it signed the Traceability Framework Agreement. The signing of the Agreement should be checked at the time of contract negotiation.

> ##### Partner 1 - Access Policy Example (Scenario 2)

Reviewer 2: Why did you put `(Scenario 2)` at the end of the headline? (same for the coming headlines as well)

Author: Markdownlint complained here that there are headlines with the same text

>     {

Reviewer 2: Code fences and indentation alignment please :-)

Author: done

        "@context": {
            "odrl": "http://www.w3.org/ns/odrl/2/"
        },
        "@type": "PolicyDefinitionRequestDto",
        "@id": "{{POLICY_ID}}",
        "policy": {
            "@type": "Policy",
            "odrl:permission": [
                {
                    "odrl:action": "USE",
                    "odrl:constraint": {
                        "@type": "LogicalConstraint",
                        "odrl:and": [
                            {
                                "@type": "Constraint",
                                "odrl:leftOperand": "BusinessPartnerNumber",
                                "odrl:operator": {
                                    "@id": "odrl:eq"
                                },
                                "odrl:rightOperand": "{{BPN6789}}"
                            }
                        ]
                    }
                }
            ]
        }
    }

##### Partner 1 - Usage/Contract Policy Example (Scenario 2)

>     {

Reviewer 2: Code fences and indentation alignment please :-)

        "policy": {
            "@type": "Policy",
            "odrl:permission": [
                {
                    "odrl:action": "USE",
                    "odrl:constraint": {
                        "@type": "LogicalConstraint",
                        "odrl:and": [
                            {
                                "@type": "Constraint",
                                "odrl:leftOperand": "FrameworkAgreement.traceability",
                                "odrl:operator": {
                                    "@id": "odrl:eq"
                                },
                                "odrl:rightOperand": "active"
                            }
                        ]
                    }
                }
            ]
        }
    }

##### Desired Outcome (Scenario 2)

Partner 2 receives the Contract Offer and is able to negotiate the contract because he owns the Traceability Credential.

#### Scenario 3

Partner 1 wants to create an Access Policy that Partner 2 can receive the Contract Offer if the BPN is matching AND Partner 2 is identified as a Dismantler (owns the "DismantlerCredential").

##### Partner 1 - Access Policy Example (Scenario 3)

>     {

Reviewer 2: Code fences and indentation alignment please :-)

                                },
                                "odrl:rightOperand": "{{BPN6789}}"
                            },
                            {
                                "@type": "Constraint",
                                "odrl:leftOperand": "Dismantler",
                                "odrl:operator": {
                                    "@id": "odrl:eq"
                                },
                                "odrl:rightOperand": "active"
                            }
                        ]
                    }
                }
            ]
        }
    }

##### Desired Outcome (Scenario 3)

Partner 2 does not receive the Contract Offer as he does not own the Dismantler Credential and is therefore not able to negotiate the contract or request the data.

#### Scenario 4

Partner 1 wants to create an Access Policy that Partner 2 can receive the Contract Offer if the BPN is matching. Further, Partner 1 only wants to create an Agreement if Partner 2 is identified as a Dismantler (owns the "DismantlerCredential"). Otherwise the Negotiation has to fail.

##### Partner 1 - Access Policy Example (Scenario 4)

>     {

Reviewer 2: Code fences and indentation alignment please :-)

        "@context": {
            "odrl": "http://www.w3.org/ns/odrl/2/"
        },
        "@type": "PolicyDefinitionRequestDto",
        "@id": "{{POLICY_ID}}",
        "policy": {
            "@type": "Policy",
            "odrl:permission": [
                {
                    "odrl:action": "USE",
                    "odrl:constraint": {
                        "@type": "LogicalConstraint",
                        "odrl:and": [
                            {
                                "@type": "Constraint",
                                "odrl:leftOperand": "BusinessPartnerNumber",
                                "odrl:operator": {
                                    "@id": "odrl:eq"
                                },
                                "odrl:rightOperand": "{{BPN6789}}"
                            }
                        ]
                    }
                }
            ]
        }
    }

##### Partner 1 - Usage/Contract Policy Example (Scenario 4)

>     {

Reviewer 2: Code fences and indentation alignment please :-)

#### Most Important Terminology

| Term     | Description |
| -------- | -------- |
| Policy | A group of one or more Rules
| Rule | An abstract concept that represents the common characteristics of Permissions, Prohibitions, and Duties
| Asset | A rsource or a collection of resources that are the subject of a Rule
| Action | An operation on an Asset
| Permission | The ability to exercise an Action over an Asset
| Prohibition | The inability to exercise an Action over an Asset
| Constraint | A boolean/logical expression that refines an Action and Asset collection or the conditions applicable to a Rule.

## Policies in Catena-X

A Catena-X policy uses the ODRL Information Model as a semantic model. Theoretically, a policy can be constructed individually using this ODRL Information Model. Catena-X does not yet use all the possibilities that ODRL provides, so the next example will show how a Catena-X policy is basically structured.

### Policy Template Example with @context: `"http://www.w3.org/ns/odrl.jsonld"`

```json
{

>   "@context": {},

Reviewer: If the context is mentioned in the heading, why is it missing here?

>   "@type": "PolicyDefinitionRequest",

Reviewer: In some examples, `PolicyDefinitionRequest` is mentioned as type. Others mention `PolicyDefinitionRequestDto`. This should be updated and aligned.

Reviewer 3: removed all occurences of `PolicyDefinitionRequestDto`

  "@id": "{{POLICY_ID}}",
  "policy": {

>     "@type": "Set",

Reviewer: The `@type` is sometimes set to `Set`, sometimes to `odrl:Set`, and sometimes to `Policy`. This should be updated and aligned.

Author: "The above example could have also used Policy type instead of Set type (**as they are equivalent**)." [Source](https://w3c.github.io/poe/model/#policy-set)

`odrl:set `is also the same. Just defined in another `@context`

    "@context": "http://www.w3.org/ns/odrl.jsonld",
    "permission": [
      {

>         "action": "use",

Reviewer 2: Also this action is not aligned. sometimes it's `use` and sometimes it's `USE`

Reviewer 2: and sometimes it's the following (see line 88):
```
        "odrl:action": {
          "@id": "odrl:use"
        },
```

Reviewer 4: The action should be `use` as described [here](https://w3c.github.io/odrl/bp/#x1-how-to-represent-a-general-permission). "USE" it will still work for backward compatibility. 

The two notation `use` or `odrl:use` in `@id` are equivalent in `JSON-LD` format, but i would stick with the prefix-less one by using the `@context`

Reviewer 3: That's right
- `{"action": "use"}`
- `{"action": {"@id": "odrl:use"}}`
- `{"action": {"@id": "http://www.w3.org/ns/odrl/2/use"}`
- and a bunch of other ways of writing it (like expanding `@id`)

are all equivalent.

`USE` however isn't as keywords, keys and values are case-sensitive in json-ld. We should add a warning for that.

Reviewer 2: Thank you! This is really good to know and should be part of the documentation ;-)

Reviewer 5: and maybe some backgrounds why it works like that is the following - I'm also just learning all that stuff... ;-)

In your jsonld context you have:
```
        "action":                {"@type": "@vocab", "@id": "odrl:action"},
        "use":                  "odrl:use",

```
The important part here is the `"@type": "@vocab"`. If you remove this, the `use` becomes a literal (string) after expanding. Having this in the jsonld context forces (or tries to force) a type on the value.
For further reference, have a look here: https://www.w3.org/TR/json-ld/#type-coercion

--
Matthias

Reviewer: There might be some problems when using `use` instead of `USE`. See https://github.com/eclipse-tractusx/tractusx-edc/issues/1048

Reviewer 3: considered.

        "constraint": [
          {

>             "@type": "LogicalConstraint",

Reviewer 5: FYI: My wish is to not have LogicalConstraints in the CX ODRL Profile for now. Because if we would have, we also needed to support or, xor, andSequence. If we don't need them (never saw an example with those...) we should NOT use LogicalConstraint and rather just have a list of (and) Constraints.

    "odrl:permission": [
      {
        "odrl:action": {
          "@id": "odrl:use"
        },
        "odrl:constraint": [
          {
            "@type": "odrl:LogicalConstraint",
            "odrl:and": [
              {
                "odrl:leftOperand": {
                  "@value": "<field>"
                },
                "odrl:operator": {
                  "@id": "odrl:eq"
                },
                "odrl:rightOperand": "<value>"
              }
            ]
          }
        ]
      }
    ]
  }
}
```

### Policies & Verifiable Credenitals (VC)

#### General Information

Catena-X uses policies to determine access to and use of data. The policies refer to verifiable credentials (VC) that are stored in the Managed Identity Wallet (MIW). Catena-X uses the principle of self-sovereign identity (SSI) here.

ℹ️ Detailed Information about SSI, VC and MIW can be found here:

- [https://github.com/eclipse-tractusx/ssi-docu](https://github.com/eclipse-tractusx/ssi-docu)
- [https://github.com/eclipse-tractusx/ssi-docu/tree/main/docs/architecture/cx-3-2](https://github.com/eclipse-tractusx/ssi-docu/tree/main/docs/architecture/cx-3-2) (for release 23.09)

> - [https://github.com/eclipse-tractusx/ssi-docu/tree/main/docs/architecture/cx-3-3](https://github.com/eclipse-tractusx/ssi-docu/tree/main/docs/architecture/cx-3-3) (for release 23.12)

Reviewer: Release 23.12 will come in a few days. This page doesn't give any information. Therefore, the link should be updated or the page should be filled with information.

Author: Unfortunately, the page is still incomplete. But I can remove the link if that avoids confusion.

Reviewer 3: Removed the whole section

The key architectural principle underlying this specification is that policy definitions must be decoupled from their corresponding VC schema. Namely, the specific **constraints** and shape of the VC schema must not be reflected in the policy definition. This allows VC schemas to be altered without impacting policy definitions.

#### Membership Constraint Example

The main difference between the various Catena-X policies is their `constraints`. For this reason, only the constraint examples are shown below instead of always showing the entire policy.

To understand what the connection between policy and VC looks like, let's take a look at the following example ([Source](https://github.com/eclipse-tractusx/ssi-docu/blob/main/docs/architecture/cx-3-2/edc/policy.definitions.md#1-membership-constraint)):

```json
{

>     "constraint": {

Reviewer: Other examples show `"@type": "Constraint"`, while this one differs.

Author: Please check line 129. It is only an excerpt from a policy.

>     "leftOperand": "Membership",

Reviewer 5: From my perspective Membership BPN should NOT be usage, but access policy. Not sure if this is worth a note here because it seems it is not clear from the context.

    "operator": "eq",
    "rightOperand": "active"
    }
}
```

An associated VC could look like this ([Source](https://github.com/eclipse-tractusx/ssi-docu/blob/main/docs/architecture/cx-3-2/3.%20Verifiable%20Credentials/Structure%20and%20Formats.md)):

```json
{
    "id": "UUID",
    "@context": [
        "https://www.w3.org/2018/credentials/v1",
        "https://w3id.org/security/suites/jws-2020/v1",
        "https://raw.githubusercontent.com/catenax-ng/product-core-schemas/main/businessPartnerData"
    ],
    "type": ["VerifiableCredential", "MembershipCredentialCX"],
    "issuanceDate": "2021-06-16T18:56:59Z",
    "expirationDate": "2022-06-16T18:56:59Z",

>     "issuer": "did", // operating environment

Reviewer: There are no comments in JSON.

Reviewer 3: maybe having "comments" that invalidate the payloads will keep readers from just copying the examples ;)

Author: This example is 1:1 from the SSI repository (see source), but I will remove the comment ;)

    ],
    "id": "<credential uid>",
    "type": [
        "VerifiableCredential",
        "SummaryCredential"
    ],
    "issuer": "<did:web:issuer>",
    "issuanceDate": "2023-06-02T12:00:00Z",
    "expirationDate": "2022-06-16T18:56:59Z",
    "credentialSubject": {
        "id": "<did:web:subject>",
        "holderIdentifier": "BPN6789",
        "items": [
        "MembershipCredential",
        "TraceabilityCredential"
        ],
        "contractTemplates": "https://public.catena-x.org/contracts/"
    }
}
```

#### Scenario 1

Partner 1 wants to create an Access Policy, that Partner 2 can only receive the Contract offer if its BPN matches AND if it has the Traceability Credential (Traceability Contract Signed). If one of those is missing, Partner 2 won't receive a Contract Offer.

##### Partner 1 - Access Policy Example

```json
    {
        "@context": {
            "odrl": "http://www.w3.org/ns/odrl/2/"
        },
        "@type": "PolicyDefinitionRequestDto",
        "@id": "{{POLICY_ID}}",
        "policy": {
            "@type": "Policy",
            "odrl:permission": [
                {
                    "odrl:action": "USE",
                    "odrl:constraint": {

>                         "@type": "LogicalConstraint",

Reviewer 5: also here, LogicalConstraint can be avoided.

        }
    }
```

Be aware, the credential within the Summary Credential is called `TraceabilityCredential`, but as the EDC does a mapping, the EDC Access Policy Entry needs to be named `FrameworkAgreement.traceability`!

##### Desired Outcome

Partner 2 receives the Contract Offer because the BPNs are matching and he owns the Traceability Credential.

#### Scenario 2

Partner 1 wants to create an Access Policy, that Partner 2 can receive the Contract Offer if its BPN matches. But a Contract Agreement should only be created if Partner 2 also signed the Traceability Framework Agreement. So in this case, Partner 2 should receive the Contract Offer in the first place, regardless if it signed the Traceability Framework Agreement. The signing of the Agreement should be checked at the time of contract negotiation.

##### Partner 1 - Access Policy Example (Scenario 2)

```json
{
    "@context": {
        "odrl": "http://www.w3.org/ns/odrl/2/"
    },
    "@type": "PolicyDefinitionRequestDto",
    "@id": "{{POLICY_ID}}",
    "policy": {
        "@type": "Policy",
        "odrl:permission": [
            {
                "odrl:action": "USE",
                "odrl:constraint": {

>                     "@type": "LogicalConstraint",

Reviewer 5: same here
